diff --git a/benchmarks/django_workload/apache-cassandra/conf/cassandra-env.sh b/benchmarks/django_workload/apache-cassandra/conf/cassandra-env.sh
index a640847..135f4b3 100644
--- a/benchmarks/django_workload/apache-cassandra/conf/cassandra-env.sh
+++ b/benchmarks/django_workload/apache-cassandra/conf/cassandra-env.sh
@@ -127,7 +127,7 @@ if [ "x$CASSANDRA_LOG_DIR" = "x" ] ; then
 fi

 #GC log path has to be defined here because it needs to access CASSANDRA_HOME
-JVM_OPTS="$JVM_OPTS -Xloggc:${CASSANDRA_LOG_DIR}/gc.log"
+# JVM_OPTS="$JVM_OPTS -Xloggc:${CASSANDRA_LOG_DIR}/gc.log"

 # Here we create the arguments that will get passed to the jvm when
 # starting cassandra.
