diff --git a/.github/scripts/fbgemm_build.bash b/.github/scripts/fbgemm_build.bash
--- a/.github/scripts/fbgemm_build.bash
+++ b/.github/scripts/fbgemm_build.bash
@@ -23,7 +23,7 @@ __configure_fbgemm_build_cmake () {
 
   # shellcheck disable=SC2206
   build_args=(
-    -DFBGEMM_USE_SANITIZER=address
+    -DFBGEMM_USE_SANITIZER=
     -DFBGEMM_LIBRARY_TYPE=${fbgemm_library_type}
     -DPYTHON_EXECUTABLE=${python_path}
   )
