diff -wbBdu -ruN '--exclude=.git' '--exclude=*.rej' '--exclude=*.orig' '--exclude=gen-cpp2' adsim_group/cpp2/client/AdSimClient.cpp build/adsim/cpp2/client/AdSimClient.cpp
--- adsim_group/cpp2/client/AdSimClient.cpp	2025-07-29 14:17:29.771582057 -0700
+++ build/adsim/cpp2/client/AdSimClient.cpp	2025-07-25 15:55:10.007667247 -0700
@@ -14,17 +14,15 @@
  * limitations under the License.
  */
 
-#include <cea/chips/adsim/cpp2/client/TLSConnect.h>
 #include <folly/SocketAddress.h>
 #include <folly/coro/BlockingWait.h>
 #include <folly/init/Init.h>
 #include <folly/io/async/EventBase.h>
 #include <folly/io/async/NotificationQueue.h>
-// clang-format off (import order matters)
-#include <cea/chips/adsim/cpp2/client/CoWorker.h>
-// clang-format on
-#include <cea/chips/adsim/if/gen-cpp2/AdSim.h>
 #include <thrift/lib/cpp2/async/RocketClientChannel.h>
+#include "CoWorker.h"
+#include "TLSConnect.h"
+#include "if/gen-cpp2/AdSim.h"
 
 DEFINE_string(host, "::1", "AdSimServer host");
 DEFINE_int32(port, 10086, "AdSimServer port");
diff -wbBdu -ruN '--exclude=.git' '--exclude=*.rej' '--exclude=*.orig' '--exclude=gen-cpp2' adsim_group/cpp2/client/CoWorker.h build/adsim/cpp2/client/CoWorker.h
--- adsim_group/cpp2/client/CoWorker.h	2025-07-29 14:17:29.772459085 -0700
+++ build/adsim/cpp2/client/CoWorker.h	2025-07-25 15:55:10.009191271 -0700
@@ -16,14 +16,16 @@
 
 #pragma once
 
-#include <cea/chips/adsim/if/gen-cpp2/AdSim.h>
 #include <fizz/client/AsyncFizzClient.h>
 #include <folly/MoveWrapper.h>
+#include <folly/coro/Task.h>
 #include <folly/futures/Promise.h>
 #include <folly/io/async/EventBase.h>
 #include <folly/io/async/NotificationQueue.h>
 #include <thrift/lib/cpp2/async/RocketClientChannel.h>
 #include <thrift/lib/cpp2/security/extensions/ThriftParametersClientExtension.h>
+#include "TLSConnect.h"
+#include "if/gen-cpp2/AdSim.h"
 
 namespace facebook::cea::chips::adsim {
 
diff -wbBdu -ruN '--exclude=.git' '--exclude=*.rej' '--exclude=*.orig' '--exclude=gen-cpp2' adsim_group/cpp2/server/AdSimServer.cpp build/adsim/cpp2/server/AdSimServer.cpp
--- adsim_group/cpp2/server/AdSimServer.cpp	2025-07-29 14:17:29.787921133 -0700
+++ build/adsim/cpp2/server/AdSimServer.cpp	2025-07-25 15:55:10.010304364 -0700
@@ -19,16 +19,17 @@
 #include <streambuf>
 #include <string>
 
-#include <cea/chips/adsim/cpp2/server/AdSimService.h>
-#include <cea/chips/adsim/cpp2/server/dwarfs/Dwarfs.h>
+#include <folly/dynamic.h>
 #include <folly/init/Init.h>
 #include <folly/io/async/test/TestSSLServer.h>
-#include <folly/json/json.h>
+#include <folly/json.h>
 #include <folly/portability/GFlags.h>
 #include <glog/logging.h>
 #include <thrift/lib/cpp2/server/Cpp2Worker.h>
 #include <thrift/lib/cpp2/server/ThriftProcessor.h>
 #include <thrift/lib/cpp2/server/ThriftServer.h>
+#include "cpp2/server/AdSimService.h"
+#include "cpp2/server/dwarfs/Dwarfs.h"
 
 DEFINE_string(config_file, "", "A json file of configurations");
 DEFINE_int32(port, -1, "Overwrite port in the config file");
diff -wbBdu -ruN '--exclude=.git' '--exclude=*.rej' '--exclude=*.orig' '--exclude=gen-cpp2' adsim_group/cpp2/server/AdSimService.cpp build/adsim/cpp2/server/AdSimService.cpp
--- adsim_group/cpp2/server/AdSimService.cpp	2025-07-29 14:17:29.788333958 -0700
+++ build/adsim/cpp2/server/AdSimService.cpp	2025-07-25 15:55:10.011314962 -0700
@@ -14,8 +14,8 @@
  * limitations under the License.
  */
 
-#include <cea/chips/adsim/cpp2/server/AdSimService.h>
-#include <cea/chips/adsim/cpp2/server/DataObjects.h>
+#include "cpp2/server/AdSimService.h"
+#include "cpp2/server/DataObjects.h"
 
 #include <fb303/ThreadCachedServiceData.h>
 #include <folly/coro/Collect.h>
diff -wbBdu -ruN '--exclude=.git' '--exclude=*.rej' '--exclude=*.orig' '--exclude=gen-cpp2' adsim_group/cpp2/server/AdSimService.h build/adsim/cpp2/server/AdSimService.h
--- adsim_group/cpp2/server/AdSimService.h	2025-07-29 14:17:29.788761875 -0700
+++ build/adsim/cpp2/server/AdSimService.h	2025-07-25 15:55:10.012430769 -0700
@@ -22,10 +22,10 @@
 #include <folly/Executor.h>
 #include <folly/MapUtil.h>
 
-#include <cea/chips/adsim/cpp2/server/DataObjects.h>
-#include <cea/chips/adsim/cpp2/server/dwarfs/Kernel.h>
-#include <cea/chips/adsim/if/gen-cpp2/AdSim.h>
 #include <fb303/BaseService.h>
+#include "cpp2/server/DataObjects.h"
+#include "cpp2/server/dwarfs/Kernel.h"
+#include "if/gen-cpp2/AdSim.h"
 
 namespace facebook::cea::chips::adsim {
 
diff -wbBdu -ruN '--exclude=.git' '--exclude=*.rej' '--exclude=*.orig' '--exclude=gen-cpp2' adsim_group/cpp2/server/DataObjects.cpp build/adsim/cpp2/server/DataObjects.cpp
--- adsim_group/cpp2/server/DataObjects.cpp	2025-07-29 14:17:29.789743832 -0700
+++ build/adsim/cpp2/server/DataObjects.cpp	2025-07-25 15:55:10.013145970 -0700
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-#include <cea/chips/adsim/cpp2/server/DataObjects.h>
+#include "cpp2/server/DataObjects.h"
 #include <folly/Singleton.h>
 
 namespace facebook::cea::chips::adsim {
diff -wbBdu -ruN '--exclude=.git' '--exclude=*.rej' '--exclude=*.orig' '--exclude=gen-cpp2' adsim_group/cpp2/server/DataObjects.h build/adsim/cpp2/server/DataObjects.h
--- adsim_group/cpp2/server/DataObjects.h	2025-07-29 14:17:29.790185910 -0700
+++ build/adsim/cpp2/server/DataObjects.h	2025-07-25 15:55:10.013816493 -0700
@@ -27,7 +27,7 @@
 #include <thrift/lib/cpp2/protocol/CompactProtocol.h>
 #include <thrift/lib/cpp2/protocol/Serializer.h>
 
-#include <cea/chips/adsim/if/gen-cpp2/AdSim.h>
+#include "if/gen-cpp2/AdSim.h"
 
 namespace facebook::cea::chips::adsim {
 
diff -wbBdu -ruN '--exclude=.git' '--exclude=*.rej' '--exclude=*.orig' '--exclude=gen-cpp2' adsim_group/cpp2/server/dwarfs/Compress.cc build/adsim/cpp2/server/dwarfs/Compress.cc
--- adsim_group/cpp2/server/dwarfs/Compress.cc	2025-07-29 14:17:29.777879647 -0700
+++ build/adsim/cpp2/server/dwarfs/Compress.cc	2025-07-25 15:55:10.014647761 -0700
@@ -18,10 +18,10 @@
 #include <util.h>
 
 #include <glog/logging.h>
-#include <cstring>
+#include <string.h>
 
-#include <cea/chips/adsim/cpp2/server/dwarfs/Compress.h>
-#include <cea/chips/adsim/cpp2/server/dwarfs/Kernel.h>
+#include "Compress.h"
+#include "Kernel.h"
 
 #include <fb303/ThreadCachedServiceData.h>
 #include <folly/coro/Task.h>
diff -wbBdu -ruN '--exclude=.git' '--exclude=*.rej' '--exclude=*.orig' '--exclude=gen-cpp2' adsim_group/cpp2/server/dwarfs/Compress.h build/adsim/cpp2/server/dwarfs/Compress.h
--- adsim_group/cpp2/server/dwarfs/Compress.h	2025-07-29 14:17:29.778403731 -0700
+++ build/adsim/cpp2/server/dwarfs/Compress.h	2025-07-25 15:55:10.015383212 -0700
@@ -21,8 +21,8 @@
 
 #include <string.h>
 
-#include <cea/chips/adsim/cpp2/server/DataObjects.h>
-#include <cea/chips/adsim/cpp2/server/dwarfs/Kernel.h>
+#include "cpp2/server/DataObjects.h"
+#include "cpp2/server/dwarfs/Kernel.h"
 
 #include <folly/coro/Task.h>
 
diff -wbBdu -ruN '--exclude=.git' '--exclude=*.rej' '--exclude=*.orig' '--exclude=gen-cpp2' adsim_group/cpp2/server/dwarfs/ConcurrentHashMap.cc build/adsim/cpp2/server/dwarfs/ConcurrentHashMap.cc
--- adsim_group/cpp2/server/dwarfs/ConcurrentHashMap.cc	2025-07-29 14:17:29.778881854 -0700
+++ build/adsim/cpp2/server/dwarfs/ConcurrentHashMap.cc	2025-07-29 14:21:45.143901510 -0700
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-#include <cea/chips/adsim/cpp2/server/dwarfs/ConcurrentHashMap.h>
+#include "ConcurrentHashMap.h"
 
 #include <fb303/ThreadCachedServiceData.h>
 #include <folly/executors/CPUThreadPoolExecutor.h>
diff -wbBdu -ruN '--exclude=.git' '--exclude=*.rej' '--exclude=*.orig' '--exclude=gen-cpp2' adsim_group/cpp2/server/dwarfs/ConcurrentHashMap.h build/adsim/cpp2/server/dwarfs/ConcurrentHashMap.h
--- adsim_group/cpp2/server/dwarfs/ConcurrentHashMap.h	2025-07-29 14:17:29.779299386 -0700
+++ build/adsim/cpp2/server/dwarfs/ConcurrentHashMap.h	2025-07-29 14:20:53.562536690 -0700
@@ -30,8 +30,8 @@
 #include <utility>
 #include <vector>
 
-#include <cea/chips/adsim/cpp2/server/DataObjects.h>
-#include <cea/chips/adsim/cpp2/server/dwarfs/Kernel.h>
+#include "cpp2/server/DataObjects.h"
+#include "cpp2/server/dwarfs/Kernel.h"
 
 #include <folly/Likely.h>
 #include <folly/SharedMutex.h>
diff -wbBdu -ruN '--exclude=.git' '--exclude=*.rej' '--exclude=*.orig' '--exclude=gen-cpp2' adsim_group/cpp2/server/dwarfs/DeepCopy.h build/adsim/cpp2/server/dwarfs/DeepCopy.h
--- adsim_group/cpp2/server/dwarfs/DeepCopy.h	2025-07-29 14:17:29.779765491 -0700
+++ build/adsim/cpp2/server/dwarfs/DeepCopy.h	2025-07-25 15:55:10.016208450 -0700
@@ -24,8 +24,8 @@
 #include <unordered_map>
 #include <vector>
 
-#include <cea/chips/adsim/cpp2/server/DataObjects.h>
-#include <cea/chips/adsim/cpp2/server/dwarfs/Kernel.h>
+#include "cpp2/server/DataObjects.h"
+#include "cpp2/server/dwarfs/Kernel.h"
 
 #include <fb303/ThreadCachedServiceData.h>
 #include <folly/container/F14Map.h>
diff -wbBdu -ruN '--exclude=.git' '--exclude=*.rej' '--exclude=*.orig' '--exclude=gen-cpp2' adsim_group/cpp2/server/dwarfs/Delay.h build/adsim/cpp2/server/dwarfs/Delay.h
--- adsim_group/cpp2/server/dwarfs/Delay.h	2025-07-29 14:17:29.780277646 -0700
+++ build/adsim/cpp2/server/dwarfs/Delay.h	2025-07-25 15:55:10.016889980 -0700
@@ -18,8 +18,8 @@
 
 #include <unistd.h>
 
-#include <cea/chips/adsim/cpp2/server/DataObjects.h>
-#include <cea/chips/adsim/cpp2/server/dwarfs/Kernel.h>
+#include "cpp2/server/DataObjects.h"
+#include "cpp2/server/dwarfs/Kernel.h"
 
 #include <fb303/ThreadCachedServiceData.h>
 #include <folly/coro/Task.h>
diff -wbBdu -ruN '--exclude=.git' '--exclude=*.rej' '--exclude=*.orig' '--exclude=gen-cpp2' adsim_group/cpp2/server/dwarfs/Dwarfs.cc build/adsim/cpp2/server/dwarfs/Dwarfs.cc
--- adsim_group/cpp2/server/dwarfs/Dwarfs.cc	2025-07-29 14:17:29.780712845 -0700
+++ build/adsim/cpp2/server/dwarfs/Dwarfs.cc	2025-07-29 03:22:28.989646826 -0700
@@ -19,19 +19,19 @@
 #include <folly/MapUtil.h>
 #include <folly/dynamic.h>
 
-#include <cea/chips/adsim/cpp2/server/dwarfs/Compress.h>
-#include <cea/chips/adsim/cpp2/server/dwarfs/ConcurrentHashMap.h>
-#include <cea/chips/adsim/cpp2/server/dwarfs/DeepCopy.h>
-#include <cea/chips/adsim/cpp2/server/dwarfs/Delay.h>
-#include <cea/chips/adsim/cpp2/server/dwarfs/Dwarfs.h>
-#include <cea/chips/adsim/cpp2/server/dwarfs/FakeIO.h>
-#include <cea/chips/adsim/cpp2/server/dwarfs/GEMM.h>
-#include <cea/chips/adsim/cpp2/server/dwarfs/HashMap.h>
-#include <cea/chips/adsim/cpp2/server/dwarfs/IBRun.h>
-#include <cea/chips/adsim/cpp2/server/dwarfs/Kernel.h>
-#include <cea/chips/adsim/cpp2/server/dwarfs/Serialize.h>
-#include <cea/chips/adsim/cpp2/server/dwarfs/Shape.h>
-#include <cea/chips/adsim/cpp2/server/dwarfs/TensorDeser.h>
+#include "Compress.h"
+#include "ConcurrentHashMap.h"
+#include "DeepCopy.h"
+#include "Delay.h"
+#include "Dwarfs.h"
+#include "FakeIO.h"
+#include "GEMM.h"
+#include "HashMap.h"
+#include "IBRun.h"
+#include "Kernel.h"
+#include "Serialize.h"
+#include "Shape.h"
+#include "TensorDeser.h"
 
 namespace facebook::cea::chips::adsim {
 
diff -wbBdu -ruN '--exclude=.git' '--exclude=*.rej' '--exclude=*.orig' '--exclude=gen-cpp2' adsim_group/cpp2/server/dwarfs/Dwarfs.h build/adsim/cpp2/server/dwarfs/Dwarfs.h
--- adsim_group/cpp2/server/dwarfs/Dwarfs.h	2025-07-29 14:17:29.781173923 -0700
+++ build/adsim/cpp2/server/dwarfs/Dwarfs.h	2025-07-25 15:55:10.019374336 -0700
@@ -19,9 +19,9 @@
 #include <string>
 
 #include <folly/MapUtil.h>
-#include <folly/json/dynamic.h>
+#include <folly/dynamic.h>
 
-#include <cea/chips/adsim/cpp2/server/dwarfs/Kernel.h>
+#include "cpp2/server/dwarfs/Kernel.h"
 
 namespace facebook::cea::chips::adsim {
 
diff -wbBdu -ruN '--exclude=.git' '--exclude=*.rej' '--exclude=*.orig' '--exclude=gen-cpp2' adsim_group/cpp2/server/dwarfs/FakeIO.h build/adsim/cpp2/server/dwarfs/FakeIO.h
--- adsim_group/cpp2/server/dwarfs/FakeIO.h	2025-07-29 14:17:29.781695752 -0700
+++ build/adsim/cpp2/server/dwarfs/FakeIO.h	2025-07-25 15:55:10.020115056 -0700
@@ -18,8 +18,8 @@
 
 #include <unistd.h>
 
-#include <cea/chips/adsim/cpp2/server/DataObjects.h>
-#include <cea/chips/adsim/cpp2/server/dwarfs/Kernel.h>
+#include "cpp2/server/DataObjects.h"
+#include "cpp2/server/dwarfs/Kernel.h"
 
 #include <fb303/ThreadCachedServiceData.h>
 #include <folly/coro/Task.h>
diff -wbBdu -ruN '--exclude=.git' '--exclude=*.rej' '--exclude=*.orig' '--exclude=gen-cpp2' adsim_group/cpp2/server/dwarfs/GEMM.cc build/adsim/cpp2/server/dwarfs/GEMM.cc
--- adsim_group/cpp2/server/dwarfs/GEMM.cc	2025-07-29 14:17:29.782197962 -0700
+++ build/adsim/cpp2/server/dwarfs/GEMM.cc	2025-07-25 15:55:10.020712910 -0700
@@ -22,17 +22,21 @@
 #include <glog/logging.h>
 
 #ifdef _OPENMP
+#include <omp.h>
 #endif
 
 #ifdef USE_MKL
+#include <mkl.h>
 #endif
 
 #include "bench/BenchUtils.h"
 #include "fbgemm/Fbgemm.h"
+#include "src/RefImplementations.h"
 #include "test/QuantizationHelpers.h"
 
-#include <cea/chips/adsim/cpp2/server/dwarfs/GEMM.h>
-#include <cea/chips/adsim/cpp2/server/dwarfs/Kernel.h>
+#include <folly/coro/Task.h>
+#include "GEMM.h"
+#include "Kernel.h"
 
 using namespace std;
 using namespace fbgemm;
diff -wbBdu -ruN '--exclude=.git' '--exclude=*.rej' '--exclude=*.orig' '--exclude=gen-cpp2' adsim_group/cpp2/server/dwarfs/GEMM.h build/adsim/cpp2/server/dwarfs/GEMM.h
--- adsim_group/cpp2/server/dwarfs/GEMM.h	2025-07-29 14:17:29.782715055 -0700
+++ build/adsim/cpp2/server/dwarfs/GEMM.h	2025-07-25 15:55:10.021362522 -0700
@@ -22,8 +22,8 @@
 
 #include <assert.h>
 
-#include <cea/chips/adsim/cpp2/server/DataObjects.h>
-#include <cea/chips/adsim/cpp2/server/dwarfs/Kernel.h>
+#include "cpp2/server/DataObjects.h"
+#include "cpp2/server/dwarfs/Kernel.h"
 
 #include <fb303/ThreadCachedServiceData.h>
 #include <folly/coro/Collect.h>
diff -wbBdu -ruN '--exclude=.git' '--exclude=*.rej' '--exclude=*.orig' '--exclude=gen-cpp2' adsim_group/cpp2/server/dwarfs/HashMap.cc build/adsim/cpp2/server/dwarfs/HashMap.cc
--- adsim_group/cpp2/server/dwarfs/HashMap.cc	2025-07-29 14:17:29.783274811 -0700
+++ build/adsim/cpp2/server/dwarfs/HashMap.cc	2025-07-25 15:55:10.022044754 -0700
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-#include <cea/chips/adsim/cpp2/server/dwarfs/HashMap.h>
+#include "HashMap.h"
 
 #include <fb303/ThreadCachedServiceData.h>
 
diff -wbBdu -ruN '--exclude=.git' '--exclude=*.rej' '--exclude=*.orig' '--exclude=gen-cpp2' adsim_group/cpp2/server/dwarfs/HashMap.h build/adsim/cpp2/server/dwarfs/HashMap.h
--- adsim_group/cpp2/server/dwarfs/HashMap.h	2025-07-29 14:17:29.783789461 -0700
+++ build/adsim/cpp2/server/dwarfs/HashMap.h	2025-07-25 15:55:10.022653935 -0700
@@ -24,8 +24,8 @@
 #include <unordered_map>
 #include <vector>
 
-#include <cea/chips/adsim/cpp2/server/DataObjects.h>
-#include <cea/chips/adsim/cpp2/server/dwarfs/Kernel.h>
+#include "cpp2/server/DataObjects.h"
+#include "cpp2/server/dwarfs/Kernel.h"
 
 #include <folly/Synchronized.h>
 #include <folly/container/F14Map.h>
diff -wbBdu -ruN '--exclude=.git' '--exclude=*.rej' '--exclude=*.orig' '--exclude=gen-cpp2' adsim_group/cpp2/server/dwarfs/IBRun.h build/adsim/cpp2/server/dwarfs/IBRun.h
--- adsim_group/cpp2/server/dwarfs/IBRun.h	2025-07-29 14:17:29.784390450 -0700
+++ build/adsim/cpp2/server/dwarfs/IBRun.h	2025-07-25 15:55:10.023200492 -0700
@@ -16,9 +16,9 @@
 
 #pragma once
 
-#include <cea/chips/adsim/cpp2/server/DataObjects.h>
-#include <cea/chips/adsim/cpp2/server/dwarfs/Kernel.h>
-#include <cea/chips/adsim/cpp2/server/dwarfs/icache_buster/ICacheBuster.h> // @manual=//cea/chips/adsim/cpp2/server/dwarfs/icache_buster:libicachebuster
+#include "cpp2/server/DataObjects.h"
+#include "cpp2/server/dwarfs/Kernel.h"
+#include "cpp2/server/dwarfs/icache_buster/ICacheBuster.h" // @manual=//cea/chips/adsim/cpp2/server/dwarfs/icache_buster:libicachebuster
 
 #include <folly/coro/Task.h>
 
diff -wbBdu -ruN '--exclude=.git' '--exclude=*.rej' '--exclude=*.orig' '--exclude=gen-cpp2' adsim_group/cpp2/server/dwarfs/icache_buster/gen_icache_buster.py build/adsim/cpp2/server/dwarfs/icache_buster/gen_icache_buster.py
--- adsim_group/cpp2/server/dwarfs/icache_buster/gen_icache_buster.py	2025-07-29 14:17:29.776967347 -0700
+++ build/adsim/cpp2/server/dwarfs/icache_buster/gen_icache_buster.py	2025-07-25 15:55:10.023937655 -0700
@@ -14,8 +14,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# pyre-unsafe
-
 """Module to split file into multiple segments."""
 
 import argparse
diff -wbBdu -ruN '--exclude=.git' '--exclude=*.rej' '--exclude=*.orig' '--exclude=gen-cpp2' adsim_group/cpp2/server/dwarfs/Kernel.cc build/adsim/cpp2/server/dwarfs/Kernel.cc
--- adsim_group/cpp2/server/dwarfs/Kernel.cc	2025-07-29 14:17:29.784831819 -0700
+++ build/adsim/cpp2/server/dwarfs/Kernel.cc	2025-07-25 16:37:11.073788021 -0700
@@ -20,7 +20,7 @@
 #include <fb303/ExportType.h>
 #include <fb303/ThreadCachedServiceData.h>
 
-#include "cea/chips/adsim/cpp2/server/dwarfs/Kernel.h"
+#include "Kernel.h"
 
 namespace facebook::cea::chips::adsim {
 
diff -wbBdu -ruN '--exclude=.git' '--exclude=*.rej' '--exclude=*.orig' '--exclude=gen-cpp2' adsim_group/cpp2/server/dwarfs/Kernel.h build/adsim/cpp2/server/dwarfs/Kernel.h
--- adsim_group/cpp2/server/dwarfs/Kernel.h	2025-07-29 14:17:29.785257753 -0700
+++ build/adsim/cpp2/server/dwarfs/Kernel.h	2025-07-25 15:55:10.025158201 -0700
@@ -21,10 +21,10 @@
 #include <unordered_map>
 #include <vector>
 
-#include <cea/chips/adsim/cpp2/server/DataObjects.h>
 #include <folly/coro/Task.h>
-#include <folly/json/dynamic.h>
+#include <folly/dynamic.h>
 #include <glog/logging.h>
+#include "cpp2/server/DataObjects.h"
 
 namespace facebook::cea::chips::adsim {
 
diff -wbBdu -ruN '--exclude=.git' '--exclude=*.rej' '--exclude=*.orig' '--exclude=gen-cpp2' adsim_group/cpp2/server/dwarfs/Serialize.h build/adsim/cpp2/server/dwarfs/Serialize.h
--- adsim_group/cpp2/server/dwarfs/Serialize.h	2025-07-29 14:17:29.785666401 -0700
+++ build/adsim/cpp2/server/dwarfs/Serialize.h	2025-07-25 15:55:10.025830557 -0700
@@ -20,15 +20,15 @@
 
 #include <algorithm>
 
-#include <cea/chips/adsim/cpp2/server/DataObjects.h>
-#include <cea/chips/adsim/cpp2/server/dwarfs/Kernel.h>
+#include "cpp2/server/DataObjects.h"
+#include "cpp2/server/dwarfs/Kernel.h"
 
 #include <folly/coro/Task.h>
 
 #include <thrift/lib/cpp2/protocol/CompactProtocol.h>
 #include <thrift/lib/cpp2/protocol/Serializer.h>
 
-#include <cea/chips/adsim/if/gen-cpp2/Serialize_types.h>
+#include "if/gen-cpp2/Serialize_types.h"
 
 namespace facebook::cea::chips::adsim {
 
diff -wbBdu -ruN '--exclude=.git' '--exclude=*.rej' '--exclude=*.orig' '--exclude=gen-cpp2' adsim_group/cpp2/server/dwarfs/Shape.h build/adsim/cpp2/server/dwarfs/Shape.h
--- adsim_group/cpp2/server/dwarfs/Shape.h	2025-07-29 14:17:29.786140568 -0700
+++ build/adsim/cpp2/server/dwarfs/Shape.h	2025-07-25 15:55:10.026656306 -0700
@@ -16,10 +16,10 @@
 
 #pragma once
 
-#include <cea/chips/adsim/cpp2/client/TLSConnect.h>
-#include <cea/chips/adsim/cpp2/server/DataObjects.h>
-#include <cea/chips/adsim/cpp2/server/dwarfs/Kernel.h>
-#include <cea/chips/adsim/if/gen-cpp2/AdSim.h>
+#include "cpp2/client/TLSConnect.h"
+#include "cpp2/server/DataObjects.h"
+#include "cpp2/server/dwarfs/Kernel.h"
+#include "if/gen-cpp2/AdSim.h"
 
 #include <fb303/ThreadCachedServiceData.h>
 #include <folly/MoveWrapper.h>
diff -wbBdu -ruN '--exclude=.git' '--exclude=*.rej' '--exclude=*.orig' '--exclude=gen-cpp2' adsim_group/cpp2/server/dwarfs/TensorDeser.cc build/adsim/cpp2/server/dwarfs/TensorDeser.cc
--- adsim_group/cpp2/server/dwarfs/TensorDeser.cc	2025-07-29 14:17:29.786570039 -0700
+++ build/adsim/cpp2/server/dwarfs/TensorDeser.cc	2025-07-29 03:10:24.727445630 -0700
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-#include <cea/chips/adsim/cpp2/server/dwarfs/TensorDeser.h>
+#include "TensorDeser.h"
 #include <cstring>
 #include <iomanip>
 #include <iostream>
diff -wbBdu -ruN '--exclude=.git' '--exclude=*.rej' '--exclude=*.orig' '--exclude=gen-cpp2' adsim_group/cpp2/server/dwarfs/TensorDeser.h build/adsim/cpp2/server/dwarfs/TensorDeser.h
--- adsim_group/cpp2/server/dwarfs/TensorDeser.h	2025-07-29 14:17:29.787024406 -0700
+++ build/adsim/cpp2/server/dwarfs/TensorDeser.h	2025-07-29 03:10:09.858016018 -0700
@@ -30,8 +30,8 @@
 #include <tuple>
 #include <vector>
 
-#include <cea/chips/adsim/cpp2/server/DataObjects.h>
-#include <cea/chips/adsim/cpp2/server/dwarfs/Kernel.h>
+#include "cpp2/server/DataObjects.h"
+#include "cpp2/server/dwarfs/Kernel.h"
 
 #include <folly/FileUtil.h>
 #include <folly/Format.h>
diff -wbBdu -ruN '--exclude=.git' '--exclude=*.rej' '--exclude=*.orig' '--exclude=gen-cpp2' adsim_group/if/AdSim.thrift build/adsim/if/AdSim.thrift
--- adsim_group/if/AdSim.thrift	2025-07-29 14:17:29.792097453 -0700
+++ build/adsim/if/AdSim.thrift	2025-07-25 15:55:10.027410947 -0700
@@ -1,4 +1,4 @@
-include "common/fb303/if/fb303.thrift"
+include "fb303.thrift"
 
 namespace cpp2 facebook.cea.chips.adsim
 namespace py3 facebook.cea.chips.adsim
diff -wbBdu -ruN '--exclude=.git' '--exclude=*.rej' '--exclude=*.orig' '--exclude=gen-cpp2' adsim_group/if/Serialize.thrift build/adsim/if/Serialize.thrift
--- adsim_group/if/Serialize.thrift	2025-07-29 14:17:29.792879556 -0700
+++ build/adsim/if/Serialize.thrift	2025-07-25 15:55:10.028022061 -0700
@@ -3,20 +3,15 @@
 cpp_include "list"
 cpp_include "folly/container/F14Map.h"
 
-include "thrift/annotation/cpp.thrift"
-
 struct SerializableInfo {
   1: bool var_bool;
   2: i32 var_i32;
   3: double var_double;
 }
 
-@cpp.Type{template = "std::list"}
-typedef list<i32> id_list_t
-@cpp.Type{template = "folly::F14VectorMap"}
-typedef map<i32, id_list_t> id_map_t
-@cpp.Type{template = "std::list"}
-typedef list<SerializableInfo> info_list_t
+typedef list<i32> (cpp.template = "std::list") id_list_t
+typedef map<i32, id_list_t> (cpp.template = "folly::F14VectorMap") id_map_t
+typedef list<SerializableInfo> (cpp.template = "std::list") info_list_t
 
 struct SerializableUnit {
   1: id_map_t map_of_list;
@@ -24,8 +19,7 @@
   3: id_list_t list_of_i32;
 }
 
-@cpp.Type{template = "std::list"}
-typedef list<SerializableUnit> unit_list_t
+typedef list<SerializableUnit> (cpp.template = "std::list") unit_list_t
 
 struct SerializableReq {
   1: i64 var_i64;
